work_queue/src/work_queue.c

```diff
@@ -7413,4 +7412,295 @@ int work_queue_specify_min_taskid(struct work_queue *q, int minid) {
+	int disk_index2 = 0;
+	for (int i = MAX_POWER_OF_TWO; i > 0; i--)
+	{
+		if (pow(2.0, i) <= (double)worker_data->disk[0])
+		{
+			disk_index1 = i;
+			break;
+		}
+	}
+	for (int i = 0; i < MAX_POWER_OF_TWO; i++)
+	{
+		if (pow(2.0, i) >= (double)worker_data->disk[worker_data->length - 1])
+		{
+			disk_index2 = i - 1;
+			break;
+		}
+	}
+	int *disk_two_values = malloc(sizeof(int) * (disk_index1 - disk_index2 + 1) * POWER_OF_TWO_DIVISIONS); // stores all 4 indicies used for rounding
+	for (int i = 0; i < (disk_index1 - disk_index2 + 1); i++)
+	{
+		for (int j = 0; j < POWER_OF_TWO_DIVISIONS; j++)
+		{
+			disk_two_values[i * POWER_OF_TWO_DIVISIONS + j] = pow(2.0, disk_index1 - i) - pow(2.0, disk_index1 - 4.0 - i) * j;
+		}
+	}
+	
+	int worker_data_index = 0;
+	for (int i = 0; i < (disk_index1 - disk_index2 + 1) * POWER_OF_TWO_DIVISIONS; i++)
+	{
+		while (worker_data->disk[worker_data_index] > disk_two_values[i])
+		{
+			if (worker_data->disk[worker_data_index] > 0) disk_losses += worker_data->disk[worker_data_index] - disk_two_values[i];
+			worker_data->disk[worker_data_index] = disk_two_values[i];
+			worker_data_index++;
+		}
+	}
+	free(disk_two_values);
+	struct work_queue_wsummary temp; // create temporary data structure to sort
+	temp.count[0] = worker_data->count[0];
+	temp.cores[0] = worker_data->cores[0];
+	temp.memory[0] = worker_data->memory[0];
+	temp.disk[0] = worker_data->disk[0];
+	temp.gpus[0] = worker_data->gpus[0];
+	temp.length = 1;
+	int should_add_entry = 1;
+	for (int i = 1; i < worker_data->length; i++)
+	{
+		should_add_entry = 1; // flag used if a new entry needs to be made
+		for (int j = 0; j < temp.length; j++)
+		{
+			if (   temp.cores[j] == worker_data->cores[i] // otherwise if one of the buckets is the same add it into the temp storage
+				&& temp.memory[j] == worker_data->memory[i]
+				&& temp.disk[j] == worker_data->disk[i]
+				&& temp.gpus[j] == worker_data->gpus[i])
+			{
+				temp.count[j] += worker_data->count[i];
+				worker_data->count[i] = 0;
+				should_add_entry = 0;
+				break;
+			}
+		}
+		if (should_add_entry) // otherwise make a new entry
+		{
+			temp.count[temp.length] = worker_data->count[i];
+			temp.cores[temp.length] = worker_data->cores[i];
+			temp.memory[temp.length] = worker_data->memory[i];
+			temp.disk[temp.length] = worker_data->disk[i];
+			temp.gpus[temp.length] = worker_data->gpus[i];
+			temp.length++;
+		}
+	}
+	int total_worker_count = 0;
+	for (int i = 0; i < temp.length; i++) // copy the temp storage back into the wsummary variable
+	{
+		worker_data->count[i] = temp.count[i];
+		worker_data->cores[i] = temp.cores[i];
+		worker_data->memory[i] = temp.memory[i];
+		worker_data->disk[i] = temp.disk[i];
+		worker_data->gpus[i] = temp.gpus[i];
+		total_worker_count += temp.count[i];
+	}
+	worker_data->length = temp.length;
+	printf("Total memory losses: %.0lfmb, Average memory loss per worker: %.2lfmb\nTotal memory available: %.0lfmb, %% of memory lost %.2lf\n",
```

Comment:
`printf` is fine while troubleshooting, but we don't want that in the final version of the library.  Could replace it with a `debug` if you want to keep it.